### PR TITLE
op-proposer: service lifecycle cleanup

### PR DESCRIPTION
**Pull request by joohhnnn**
In reference to [pull/7682](https://github.com/ethereum-optimism/optimism/pull/7682), I've addressed portions of issue [7671](https://github.com/ethereum-optimism/optimism/issues/7671) / [7684](https://github.com/ethereum-optimism/optimism/issues/7684), within the proposer code. I've strived to maintain a coding style consistent with pull/7682, though I'm not certain if anything has been overlooked.

### CODE REVIEW

**Comment by protolambda** [comment]
Thank you, this refactor looks great. I'm catching up with some more PR review after having been OOO for a while, but will take a look at this soon!

File: op-proposer/metrics/noop.go [hunk 1]
```diff
@@ -17,3 +24,7 @@ func (*noopMetrics) RecordInfo(version string) {}
 func (*noopMetrics) RecordUp()                 {}
 
 func (*noopMetrics) RecordL2BlocksProposed(l2ref eth.L2BlockRef) {}
+
+func (*noopMetrics) StartBalanceMetrics(log.Logger, *ethclient.Client, common.Address) io.Closer {
+	return nil
+}
```
Comment (trianglesphere) [inline op-proposer/metrics/noop.go]: Needs a newline at the end of the file.

File: op-proposer/proposer/driver.go [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+
+type RollupClient interface {
+	SyncStatus(ctx context.Context) (*eth.SyncStatus, error)
+	OutputAtBlock(ctx context.Context, blockNum uint64) (*eth.OutputResponse, error)
+}
+
+type DriverSetup struct {
+	Log      log.Logger
+	Metr     metrics.Metricer
+	Cfg      ProposerConfig //what need to be contented
```
Comment (trianglesphere) [inline op-proposer/proposer/driver.go]: what is "contented"

File: op-proposer/proposer/service.go [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+	return result
+}
+
+var _ cliapp.Lifecycle = (*ProposerService)(nil)
+
+// Driver returns the handler on the L2Output-submitter driver element,
+// to start/stop/restart the L2Output-submission work, for use in testing.
+func (ps *ProposerService) Driver() rpc.ProposerDriver {
+	return ps.driver
+}
```
Comment (trianglesphere) [inline op-proposer/proposer/service.go]: Ditto newline at the end

**Review by trianglesphere — COMMENTED** [review]
Structure looks good, but this needs several golint fixes. run `make lint-go` from the top level of the repo to verify that they are fixed. Best practice is also to split up moving code from modifying code - it's not easy to quickly check that the function for `l2_output_submitter.go` got correctly moved over to `driver.go`

Adding the Admin RPC is also very nice, but again, a little harder to review when including it with the other refactor.

```
op-e2e/faultproof_test.go:535:55: Error return value of `(github.com/ethereum-optimism/optimism/op-proposer/proposer/rpc.ProposerDriver).StopL2OutputSubmitting` is not checked (errcheck)
        sys.L2OutputSubmitter.Driver().StopL2OutputSubmitting()
                                                             ^
op-e2e/system_fpp_test.go:262:55: Error return value of `(github.com/ethereum-optimism/optimism/op-proposer/proposer/rpc.ProposerDriver).StopL2OutputSubmitting` is not checked (errcheck)
        sys.L2OutputSubmitter.Driver().StopL2OutputSubmitting()
                                                             ^
op-proposer/metrics/noop.go:30: File is not `goimports`-ed (goimports)
}
op-proposer/proposer/config.go:9: File is not `goimports`-ed (goimports)

op-proposer/proposer/driver.go:12: File is not `goimports`-ed (goimports)
```

File: op-proposer/proposer/driver.go (hunk 2 above)
Comment (joohhnnn) [reply op-proposer/proposer/driver.go]: sorry, forget to clean this draft mark

**Comment by joohhnnn** [comment]
> Best practice is also to split up moving code from modifying code

Thank you @trianglesphere  for the advice. Next time, I will open two separate pull requests to make the review process smoother :)

**Comment by trianglesphere** [comment]
@joohhnnn there's a couple more lints & one merge conflict to fix.

**Comment by ajsutton** [comment]
@trianglesphere Are you ok to do a final review on this or do you want me to dive in and review (I'd have to come up to speed on the general context of the proposer)?

**Review by trianglesphere — APPROVED** [review]

**Pull request merged**